src/include/kdbio.h

```diff
@@ -0,0 +1,495 @@
+ * @param  timerOp  file descriptor operation
+ * @return          callback
+ */
+ElektraIoTimerCallback elektraIoTimerGetCallback (ElektraIoTimerOperation * timerOp);
+
+/**
+ * Create a new idle operation.
+ * Free returned data after usage.
+ *
+ * @param  enabled     0 to disable, any other value for enabled
+ * @param  callback    Called when file descriptor state has changes
+ * @param  data        Custom private data
+ * @return             file descriptor operation structure
+ */
+ElektraIoIdleOperation * elektraIoNewIdleOperation (int enabled, ElektraIoIdleCallback callback, void * data);
+
+/**
+ * Enable or disable idle operation
+ * @param  idleOp  idle operation
+ * @param  enabled 0 to disabled, any other value for enabled
+ * @return         0 on success, any other value on error
+ */
+int elektraIoIdleSetEnabled (ElektraIoIdleOperation * idleOp, int enabled);
+
+/**
+ * Check if idle operation is enabled or disabled
+ * @param  idleOp   idle operation
+ * @return         0 to disabled, any other value for enabled
+ */
+int elektraIoIdleIsEnabled (ElektraIoIdleOperation * idleOp);
+
+/**
+ * Set private binding data for operation
+ * @param  idleOp   idle operation
+ * @param  data     pointer to data
+ * @return          0 on success, any other value on error
+ */
+int elektraIoIdleSetBindingData (ElektraIoIdleOperation * idleOp, void * data);
+
+/**
+ * Get private binding data from operation
+ * @param  idleOp   idle operation
+ * @return          pointer to data or NULL on error
+ */
+void * elektraIoIdleGetBindingData (ElektraIoIdleOperation * idleOp);
+
+/**
+ * Get private data from operation
+ * @param  idleOp   idle operation
+ * @return          pointer to data or NULL on error
+ */
+void * elektraIoIdleGetData (ElektraIoIdleOperation * idleOp);
+
+/**
+ * Get callback of idle operation
+ * @param  idleOp   idle operation
+ * @return          callback
+ */
+ElektraIoIdleCallback elektraIoIdleGetCallback (ElektraIoIdleOperation * idleOp);
+
+/**
+ * Create and initialize IO-Binding binding.
+ * Used by elektraIoBindingTestSuite between tests to get a fresh binding instance
+ * @return initialized IO-Binding binding
+ */
+typedef ElektraIoInterface * (*ElektraIoTestSuiteCreateBinding) (void);
+
+/**
+ * Start io processing (e.g. event loop)
+ * Used by elektraIoBindingTestSuite.
+ *
+ * Should not return until processing is stopped (e.g. by calling ElektraIoTestSuiteStop)
+ */
+typedef void (*ElektraIoTestSuiteStart) (void);
+
+/**
+ * Stop IO processing (e.g. event loop)
+ * Used by elektraIoTestSuite
+ */
+typedef void (*ElektraIoTestSuiteStop) (void);
+
+/**
+ * Test-Suite for IO-Bindings.
+ *
+ * @param createBinding Create and initialize a new binding instance
+ * @param start         Pointer to the start function
+ * @param stop          Pointer to the stop function
+ */
+void elektraIoTestSuite (ElektraIoTestSuiteCreateBinding createBinding, ElektraIoTestSuiteStart start, ElektraIoTestSuiteStop stop);
```

Comment:
should be in a separate header file, or somewhere internally.

Reply:
I inteded for it to be public so that application developers can test custom I/O binding implementations against the test suite. A different file would be better though (e.g kdbiotest.h)